### PR TITLE
Additional config options for wait_for status method.

### PR DESCRIPTION
**Pull request by toha10**
Amount of retries and checking interval was added as additional options for 'wait_for_status' method of compute, image and storage resources. It provides invidual and flexible settings for every listed service for checking statuses of their objects, like instance, snapshot, volume.

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test FAILed.   http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/15/

File: cfglib.py [hunk 1]
```diff
@@ -300,7 +325,12 @@
     cfg.BoolOpt('convert_to_raw', default='True',
                 help='convert to raw images'),
     cfg.StrOpt('backend', default='file',
-               help='backend for images')
+               help='backend for images'),
+    cfg.IntOpt('wait_for_status_retries', default=60,
+               help='Number of retries for checking nova service object'),
+    cfg.IntOpt('wait_for_status_interval', default=3,
+               help='Time period in seconds between checking status'
+                    'of nova service object')
```
Comment (roman-verchikov) [inline cfglib.py]: Can we make those options global? For example:
```
retries_group = cfg.OptGroup('retries')
retry_opts = [
   cfg.IntOpt('wait_for_status_retries', default=60),
   cfg.IntOpt('wait_for_status_interval', default=3),
   cfg.ListOpt('retry_services_list', default=['nova_compute', 'glance_image', 'cinder_storage'])
]
```

File: cloudferrylib/os/image/glance_image.py [hunk 2]
```diff
@@ -214,8 +215,26 @@ def deploy(self, info, callback=None):
...
+        if limit_retry <= 0:
+            LOG.warn("Treating negative or zero config value %s "
+                     "for 'wait_for_status_retries' of image service."
+                     % limit_retry)
+            limit_retry = 60
+        if retry_interval <= 0:
+            LOG.warn("Treating negative or zero config value %s "
+                     "for 'wait_for_status_interval' of image service."
+                     % retry_interval)
+            retry_interval = 3
```
Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: Retry logic is identical for any task. Lets move it to a separate module and use it from the tasks in question.

**Comment by roman-verchikov** [comment]
1. Move introduced options to a global config space;
2. Move retry logic to a specific class, so that it's not scattered around the codebase;
3. Fix broken unit tests
4. Cover introduced functionality with unit tests

**Pull request closed without merging**